src/dmd/root/filename.d

```diff
@@ -894,4 +895,80 @@ version(Windows)
 
         return F(ret);
     }
+
+}
+
+version(Posix)
+{
+
+    /**************************
+    Takes a callable F and applies it to the result of converting
+    `fileName` to an absolute file path (char*)
+    Params:
+        fileName = The file name to be converted to an absolute path
+    Returns: Whatever `F` returns.
+    */
+    auto absPathThen(alias F)(const(char)* fileName)
+    {
+        import core.sys.posix.stdlib: realpath, free;
+        auto absPath = realpath(fileName, null /* realpath allocates */);
+        scope(exit) free(absPath);
+        return F(absPath);
+    }
+}
+else
+{
+    /**************************
+    Takes a callable F and applies it to the result of converting
+    `fileName` to an absolute file path (char*)
+    Params:
+        fileName = The file name to be converted to an absolute path
+    Returns: Whatever `F` returns.
+     */
+    auto absPathThen(alias F)(const(char)* fileName)
+    {
+        import core.sys.windows.winnls: WideCharToMultiByte;
+        import core.stdc.stdlib: malloc, free;
+
+        return fileName.extendedPathThen!((wpath) {
+                // first find out how long the buffer must be to store the result
+                const length = WideCharToMultiByte(0,    // code page
+                                                   0,    // flags
+                                                   wpath,
+                                                   -1,   // wpath len, -1 is null terminated
+                                                   null, // multibyte output ptr
+                                                   0,    // multibyte output length
+                                                   null, // default char
+                                                   null, // if used default char
+                );
+
+                if(!length) return F((char*).init);
+
+                char[1024] buf;
```

Comment:
= void?

Reply:
Why?

Reply:
It's a minor optimization.  No need to default-initialize it if you're just going to overwrite it.